x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/DeprecationCheckerComponents.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.deprecation;
+
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+
+public class DeprecationCheckerComponents {
```

Comment:
Not necessary but a super-minor cleanup we could do is move this into the `DeprecationChecker` interface as a sibling of `CheckResult`. 

(I forgot classes could be defined inside interfaces when I said to move this to its own file. Again, feel free to disregard this)